Automattic/wp-calypso
# A4A Dev Sites: Render site preview link expiration copy.

Fixes https://github.com/Automattic/dotcom-forge/issues/9169

## Proposed Changes

* Based on the implementation of the `SitePreviewLink` component, it actually is designed to display a list of preview link components (not 1 preview link but multiple). Therefore I am renaming the component to plural `SitePreviewLinks`.
* I am separating out the SitePreviewLink to a separate component for code readability. 
* The site preview links API now returns also `expires_at` preview link property, I am uptaking this fact.
* If the `expires_at` prop is present, we calculate the remaining days and hours and display the copy to inform the user.

## Why are these changes being made?


* Provide an informative copy to the user about the site preview link expiration time.

## Testing Instructions



#### Pre-requisits
- An agency account with:
  - At least one, free A4A development site created.
  - At least one, agency managed, "Coming soon" non-development site created.
- Apply the PR

#### Testing: The Happy Path
1. Go to https://calypso.localhost:3000/sites?flags=a4a-dev-sites.
2. Select your free A4A development site.
3. Select "Share a preview link with clients".
![Screenshot 2024-09-23 at 14 21 31](https://github.com/user-attachments/assets/dd352fee-d73b-4916-9831-a4e52a7496d4)
4. In the modal, enable "Share site" toggle. 
5. Ensure the expiration copy is displayed.

#### Testing: Links Without Expiry
1. Go to https://calypso.localhost:3000/sites?flags=a4a-dev-sites.
2. Select your agency-managed, non-development, "Coming soon" site.
3. Expand the site actions (ellipsis button).
4. Select "Share site for preview".
5. In the modal, enable "Share site".
6. Ensure the expiration copy is not displayed.

The site preview links is also available in the privacy settings of "Coming soon" non-development sites.

## Pre-merge Checklist




| Before | After |
|:---:|:---:|
| <img src="https://github.com/user-attachments/assets/10f8d830-f68c-4167-b72e-702a85b691e9" width="400" style="max-height:200px;"> | <img src="https://github.com/user-attachments/assets/ceef93f3-6baa-48fc-a9cf-c3017a913a74" width="400" style="max-height:200px;"> |
